mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala

```diff
@@ -114,10 +118,39 @@ class ScalaToplevelMtags(
             )
           } else
             loop(indent, false, currRegion, newExpectTemplate)
+        case IDENTIFIER
```

Comment:
Should this not show up in `ToplevelSuite` ? It would be cool to test it in the expect suites

Reply:
There is also `ScalaTopLevelSuite`.

Overall looks good. I only noticed that with `includeInnerClasses = true` it might return inner methods.
I added the following test in ScalaTopLevelSuite with fails:
```scala
  check(
    "toplevel-extesion".only,
    """|package a
       |
       |extension (s: String)
       |  def foo: Int = ???
       |  def bar: String =
       |    def hmm: Int = ???  // <- shouldn't be returned
       |    ???
       |
       |""".stripMargin,
    List(
      "a/",
      "a/Test$package.",
      "a/Test$package.bar.",
      "a/Test$package.foo.",
    ),
    all = true,
  )
```
Output:
```
- tests.ScalaToplevelSuite:
  * tests.ScalaToplevelSuite.toplevel-extesion - munit.ComparisonFailException: /home/dos65/projects/metals/samples/inc_error/metals/tests/unit/src/test/scala/tests/ScalaToplevelSuite.scala:250
249:
250:  check(
251:    "toplevel-extesion".only,
diff assertion failed
=> Obtained
    """|a/
       |a/Test$package.
       |a/Test$package.bar.
       |a/Test$package.foo.
       |a/Test$package.hmm.
       |""".stripMargin
``` 


However, I can't remember when `includeInnerClasses=true` is used.
Also, unfortunately, I don't remember what was the purpose in returning toplevel DEF/VAL/EXTENSION and others properly. :smiling_face_with_tear: 
In theory, the only toplevel that is required for symbols search is `${packageName}/${sourceName}$package.`
I'm wondering, if `Import missing` action would work if `extension` kw would  just register that there is a toplevel owner without parsing inner Def :thinking:

Reply:
> I'm wondering, if Import missing action would work if extension kw would just register that there is a toplevel owner without parsing inner Def 🤔

It would be great, but I think it's impossible because

- extension method callee doesn't know what import will make the extension method call possible
  - the only clue is the name of the extension method
  - if we expand this feature into extension method auto-completion, we need to know (index) the method name

Reply:
I added tests and fix the issue @dos65 mentioned in https://github.com/scalameta/metals/pull/4141/commits/91a3903c791aaffa2a975f21a7f0f8ad964ca609 👍

> Should this not show up in ToplevelSuite ? It would be cool to test it in the expect suites

I don't think it appears in ToplevelSuite, as `ToplevelSuite` generate by `Mtags.toplevels` that never parse the inner classes or extension methods, not `Mtags.allToplevels`.
Should we add expect suite for allToplevels ?

Reply:
Ach, ok I looked again into the test suites and realised that we have the additional `innerClasses` boolean option. No need to tests in ToplevelSuite